Bluetooth: controller: Add association between adv, sync and iso

cvinayak opened the pull request:
Add association between extended advertising, periodic
advertising and broadcast ISO instance, so that attempts to
create and terminate BIG can detect error conditions.

Error conditions being, trying to create BIG without a valid
periodic advertising train, or terminating BIG without prior
creation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Add association between adv, sync and iso

Add association between extended advertising, periodic
advertising and broadcast ISO instance, so that attempts to
create and terminate BIG can detect error conditions.

Error conditions being, trying to create BIG without a valid
periodic advertising train, or terminating BIG without prior
creation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_adv_iso.c:
```diff
@@ -67,7 +69,8 @@ uint8_t ll_big_create(uint8_t big_handle, uint8_t adv_handle, uint8_t num_bis,
 	 * with another BIG.
 	 */
-	if (!adv || !adv->lll.sync || adv->lll.sync->adv_iso) {
+	lll_adv_sync = adv->lll.sync;
+	if (!adv || !lll_adv_sync || lll_adv_sync->iso) {
```
Either the `!adv` is unneeded, or else it need to come before the dereference.

cvinayak replied:
hmm... i had fixed it I thought, may be I messed up with merge conflicts when rebasing the draft PR... will fix it.

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_adv_iso.c:
```diff
@@ -136,13 +139,21 @@ uint8_t ll_big_create(uint8_t big_handle, uint8_t adv_handle, uint8_t num_bis,
 	node_rx->hdr.handle = big_handle;
 	node_rx->hdr.rx_ftr.param = adv_iso;
 
+	/* FIXME: instead of is_created use lll_adv_iso->adv being set */
```
Is anything stopping you from doing that in this PR? Sounds like a simple change.

cvinayak replied:
Will do.

cvinayak committed:
Bluetooth: controller: Use Ext. Advertising association for BIG created

Check for associated Extended Advertising set to determine
whether Broadcast ISO Group instance has been already
created.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley reviewed (approved):
LGTM
